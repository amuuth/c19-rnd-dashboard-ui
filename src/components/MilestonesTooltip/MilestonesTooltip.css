.milestones.tooltip {
  color: var(--color);
  background-color: var(--bg);
  position: absolute;
  bottom: 3em;
  padding: 0.5em;
  border: 0.02em solid var(--color);
  width: 9em;
}
