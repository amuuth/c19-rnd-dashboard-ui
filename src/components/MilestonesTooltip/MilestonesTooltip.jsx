import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'
import './MilestonesTooltip.css'

const propTypes = {
  startDate: PropTypes.string.isRequired,
  endDate: PropTypes.string.isRequired,
  className: PropTypes.string,
}

const defaultProps = {
  className: '',
}

const dateFormat = 'MMM Do, YY'

export const MilestonesTooltip = ({ className, startDate, endDate }) => {
  const _startDate = moment(startDate)
  const _endDate = moment(endDate)
  return (
    <div className={`tooltip milestones ${className}`}>
      <span className='pointer'></span>
      {`${_startDate.format(dateFormat)} - ${_endDate.format(dateFormat)}`}
    </div>
  )
}

MilestonesTooltip.propTypes = propTypes
MilestonesTooltip.defaultProps = defaultProps
