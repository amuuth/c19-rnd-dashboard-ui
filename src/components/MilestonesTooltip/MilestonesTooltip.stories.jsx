import React from 'react'
import { MilestonesTooltip } from './MilestonesTooltip'

export default {
  title: 'Tooltips',
}

export const Milestones = () => (
  <MilestonesTooltip
    startDate={new Date('2020/02/10').toISOString()}
    endDate={new Date('2020/03/05').toISOString()}
  />
)
