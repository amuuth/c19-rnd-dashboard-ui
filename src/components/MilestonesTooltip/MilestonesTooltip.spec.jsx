import React from 'react'
import { shallow } from 'enzyme'
import { MilestonesTooltip } from './MilestonesTooltip'

it('should render a tooltip component', () => {
  const wrapper = shallow(
    <MilestonesTooltip
      startDate={new Date('2020/02/20').toISOString()}
      endDate={new Date('2020/03/05').toISOString()}
    />
  )
  expect(wrapper.find('.tooltip').exists()).toBe(true)
  expect(wrapper).toMatchSnapshot()
})
