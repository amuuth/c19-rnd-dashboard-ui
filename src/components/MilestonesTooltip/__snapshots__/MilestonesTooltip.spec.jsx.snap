// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a tooltip component 1`] = `ShallowWrapper {}`;
